sky/clouds/gcp.py

```diff
@@ -187,6 +190,31 @@ def get_egress_cost(self, num_gigabytes):
     def is_same_cloud(self, other):
         return isinstance(other, GCP)
 
+    def get_image_size(self, image_id: str, region: Optional[str]) -> float:
+        del region  # unused
+        if image_id.startswith('skypilot:'):
+            return DEFAULT_GCP_IMAGE_GB
+        try:
+            compute = gcp.build('compute',
+                                'v1',
+                                credentials=None,
+                                cache_discovery=False)
+        except gcp.credential_error_exception() as e:
+            return DEFAULT_GCP_IMAGE_GB
+        try:
+            image_attrs = image_id.split('/')
+            project = image_attrs[1]
+            image_name = image_attrs[-1]
+            image_infos = compute.images().get(project=project,
+                                               image=image_name).execute()
+            return float(image_infos['diskSizeGb'])
+        except gcp.http_error_exception() as e:
+            if e.resp.status == 404 and 'was not found' in e.reason:
```

Comment:
Do we need to explicitly check `'was not found' in e.reason`? Wouldn't  e.resp.status == 404 be sufficient?

Reply:
Good point! Also found another 403 error which indicates we fail to access the image.